tests/components/homematicip_cloud/helper.py

```diff
@@ -23,26 +30,36 @@ def get_and_check_entity_basics(
     """Get and test basic device."""
     ha_entity = hass.states.get(entity_id)
     assert ha_entity is not None
-    assert ha_entity.attributes["model_type"] == device_model
+    if device_model:
+        assert ha_entity.attributes[ATTR_MODEL_TYPE] == device_model
     assert ha_entity.name == entity_name
 
-    hmip_device = default_mock_hap.home.template.search_mock_device_by_id(
-        ha_entity.attributes["id"]
-    )
-    assert hmip_device is not None
+    hmip_device = default_mock_hap.hmip_device_by_entity_id.get(entity_id.lower())
```

Comment:
Why do we need to use `lower`? Isn't the entity_id already lowercase?

Reply:
Agree, can be removed. I check my test data.